tasks/Qc.wdl

```diff
@@ -278,25 +278,31 @@ task CollectDuplicateMetrics {
   }
 }
 
-# Build BAM/CRAM index
+# Build BAM/CRAM index, this should run before all other QC tasks
 task BuildBamIndex {
   input {
     File input_bam
+    String base_name
+    Boolean is_bam
     File ref_cache
     Int preemptible_tries
   }
 
   Float ref_size = size(ref_cache, "GiB") * 5 + 1.0
   Int disk_size = ceil(size(input_bam, "GiB") + ref_size) + 20
+  String bam_link = base_name + "." + if is_bam then "bam" else "cram"
```

Comment:
If we did want to try the string manipulation route, maybe something like:
```wdl
String bam_link = sub(basename(input_bam), basename(basename(input_bam, ".bam"), ".cram"), base_name )
```

Basically, the idea is:
1. Get the basename of the file with the extension still attached
2. Get the basename of the file without an extension (regardless of if it's `.bam` or `.cram`)
3. Substitute the extensionless-basename with our new `base_name` in the version with the extension still attached.

Reply:
Do you know how basename will work with files named like `foo.cram.bar.cram`? I would assume the behavior to return `foo.cram.bar` but curious if you've tried this before. I agree that one less input would be best. My only concern are edge cases that I've not though of since it's impossible to know how files will be named when ingested to AnVIL.

Reply:
Hopefully the testing bears this out, but it should only remove the extension specified in the call to `basename` (or include all of them if nothing is supplied).

Reply:
This works for a CRAM with multiple .cram strings!

Input:
```
gs://fc-59e889f9-1669-496d-ae1a-1a3322c63e6c/CCDG_13607_Project_CCDG_13607_B01_GRM_WGS.cram.2019-02-06_Sample_HG00096_analysis_HG00096.final.cram
```

Output:
```
gs://fc-59e889f9-1669-496d-ae1a-1a3322c63e6c/a913a74c-b83f-48d4-9097-95b0b7413045/WholeGenomeSingleSampleQc/6371e21f-7fc3-4a38-8e26-cb8fc5c808db/call-BuildBamIndex/HG00096.cram
```

We'll find out in later testing if it works on BAM equally well, but I would assume so and plan on merging after a few commits to remove `is_bam` as an input.